yb-voyager/src/tgtdb/event.go

```diff
@@ -58,6 +58,31 @@ func (e *Event) String() string {
 		e.Vsn, e.Op, e.SchemaName, e.TableName, mapStr(e.Key), mapStr(e.BeforeFields), mapStr(e.Fields), e.ExporterRole)
 }
 
+func (e *Event) Copy() Event {
+	keyCopy := make(map[string]*string, len(e.Key))
+	for k, v := range e.Key {
+		keyCopy[k] = v
+	}
+	fieldsCopy := make(map[string]*string, len(e.Fields))
+	for k, v := range e.Fields {
+		fieldsCopy[k] = v
+	}
+	beforeFieldsCopy := make(map[string]*string, len(e.BeforeFields))
+	for k, v := range e.BeforeFields {
+		beforeFieldsCopy[k] = v
+	}
```

Comment:
```
idFn := func(k string, v *string) (string, *string) {return k, v}
keyCopy := lo.MapEntries(e.Key, idFn)
fieldsCopy := lo.MapEntries(e.Fields, idFn)
beforeFieldsCopy := lo.MapEntries(e.BeforeFields, idFn)
```
Alternatively, you can simply drop the intermediate variables `keyCopy`, `fieldsCopy`, etc. and directly use `MapEntries` in the `Event` Initialiser.
```
return Event {
  ...
  Key: lo.MapEntries(ev.Key, idFn),
  Fields: lo.MapEntries(e.Fields, idFn),
  ...
}
```